frb_dart/lib/src/isolate/web.dart

```diff
@@ -0,0 +1,167 @@
+  /// The receive port.
+  final RawReceivePort port;
+
+  static dynamic _extractData(MessageEvent event) => event.data;
+
+  /// Create a new receive port from an optional [RawReceivePort].
+  ReceivePort([RawReceivePort? port]) : port = port ?? RawReceivePort();
+
+  @override
+  StreamSubscription listen(
+    void Function(dynamic event)? onData, {
+    Function? onError,
+    void Function()? onDone,
+    bool? cancelOnError,
+  }) {
+    return port.receivePort.onMessage.map(_extractData).listen(
+          onData,
+          onError: onError,
+          onDone: onDone,
+          cancelOnError: cancelOnError,
+        );
+  }
+
+  /// The send port.
+  SendPort get sendPort => port.sendPort;
+
+  /// Close the receive port, ignoring any further messages.
+  void close() => port.receivePort.close();
+}
+
+ReceivePort broadcastPort(String channelName) =>
+    ReceivePort(RawReceivePort(Channel.broadcastChannel(channelName)));
+
+/// [html.MessagePort]'s interface.
+abstract class PortLike extends EventTarget {
+  factory PortLike.messagePort(html.MessagePort port) = _MessagePortWrapper;
+  factory PortLike.broadcastChannel(BroadcastChannel channel) =
+      _BroadcastPortWrapper;
+  void postMessage(Object? value);
+  void close();
+  NativePortType get nativePort;
+}
+
+class _MessagePortWrapper implements PortLike, html.MessagePort {
+  final html.MessagePort port;
+  _MessagePortWrapper(this.port);
+
+  @override
+  get nativePort => port;
+  @override
+  void addEventListener(String type, html.EventListener? listener,
+          [bool? useCapture]) =>
+      port.addEventListener(type, listener, useCapture);
+  @override
+  bool dispatchEvent(html.Event event) => port.dispatchEvent(event);
+  @override
+  void removeEventListener(String type, html.EventListener? listener,
+          [bool? useCapture]) =>
+      port.removeEventListener(type, listener, useCapture);
+  @override
+  void postMessage(message, [List<Object>? transfer]) =>
+      port.postMessage(message, transfer);
+  @override
+  void close() => port.close();
+  @override
+  html.Events get on => port.on;
+  @override
+  Stream<html.MessageEvent> get onMessage => port.onMessage;
+}
+
+class _BroadcastPortWrapper implements PortLike, html.BroadcastChannel {
+  final html.BroadcastChannel channel;
+  _BroadcastPortWrapper(this.channel);
+
+  @override
+  get nativePort => channel;
+  @override
+  void addEventListener(String type, html.EventListener? listener,
+          [bool? useCapture]) =>
+      channel.addEventListener(type, listener, useCapture);
+  @override
+  bool dispatchEvent(html.Event event) => channel.dispatchEvent(event);
+  @override
+  void removeEventListener(String type, html.EventListener? listener,
+          [bool? useCapture]) =>
+      channel.removeEventListener(type, listener, useCapture);
+  @override
+  void postMessage(message, [List<Object>? transfer]) =>
+      channel.postMessage(message ?? false);
```

Comment:
nit: Why `??false` here? And `transfer` does not seem to be passed to the channel

Reply:
This presents a limitation of BroadcastChannel, that is it cannot carry transferables and will unconditionally clone the items. I don't remember if the native BroadcastChannel would accept nulls or not, might have to double check.

Reply:
Would be great to at least move this function onto top of the class, with a big code comment about the fact (possibly copying your comments here). Otherwise, when looking at it, I originally thought this is nothing but the "proxy design pattern" so everything is perfectly forwarded.

Even better, we can extract the parts that perfectly forward functions into a `mixin`.